src/Nethermind/Nethermind.Merge.Plugin/GC/GCKeeper.cs

```diff
@@ -29,7 +30,7 @@ public GCKeeper(IGCStrategy gcStrategy, ILogManager logManager)
     public IDisposable TryStartNoGCRegion(long? size = null)
     {
         size ??= _defaultSize;
-        if (_gcStrategy.CanStartNoGCRegion())
+        if (GCScheduler.MarkGCPaused() && _gcStrategy.CanStartNoGCRegion())
```

Comment:
So if there is a GC already we won't start NoGCRegion, right?

Reply:
Wasn't my intention. Instead am now capturing the return and if it was true unpause in Dispose, but don't prevent NoGCRegion